### PR TITLE
Serve gzipped static assets when Content-Encoding is provided and matches

### PR DESCRIPTION
**Pull request by jkongie**

### CODE REVIEW

File: core/web/middleware.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	fpath := path.Clean(upath)
+	if strings.HasSuffix(fpath, "/") {
+		http.NotFound(w, r)
+		return
+	}
+
+	// Find the best acceptable file, including trying uncompressed
+	if file, info, err := f.findBestFile(w, r, fpath); err == nil {
+		http.ServeContent(w, r, fpath, info.ModTime(), file)
+		// Satisy errcheck
```
Comment (samsondav) [inline core/web/middleware.go:172]: Satisfy*

Comment (samsondav) [reply core/web/middleware.go:172]: Why not something like `logger.ErrorIfCalling(file.Close)`

**Review by samsondav — APPROVED** [review]

File: core/web/middleware.go (hunk 1 above)
Comment (jkongie) [reply core/web/middleware.go]: Fixed

**Comment by jkongie** [comment]
Updated to fix some merge conflicts and Sam's suggestion. Ready for re-review

**Review by samsondav — APPROVED** [review]

**Pull request merged**